GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -155,7 +155,7 @@ public bool TryKillRunningProcess(out string processName, out int exitCode, out
             }
         }
 
-        public virtual void RejectCredentials(string repoUrl, string username = null, string password = null)
+        public virtual bool TryRejectCredentials(ITracer tracer, string repoUrl, string username, string password, out string errorMessage)

```

Comment:
Most of the methods in `GitProcess` return the `Result` and let the caller decide what to do.  What are you thoughts on using that here and letting the `CredentialManager` handle the result?

Reply:
I could do that - would that be enough (along with `CredentialManager` keeping an `ITracer` member variable) to justify a `CredentialManager` class?

I liked the API you proposed for `ICredentialManager`, and while what we have now with `Try*` methods isn't quite there, I think it is closer to the ideal goal than returning `GitProcess.Result` (i.e. - If we are going to remove the `CredentialManager` class, I would rather keep this API than transforming it to return a `GitProcess.Result`).